benchmarks/cudf-merge.py

```diff
@@ -0,0 +1,325 @@
+                "payload": cupy.arange(local_size, dtype="int64"),
+            }
+        )
+    return df
+
+
+async def worker(rank, eps, args):
+    # Setting current device and make RMM use it
+    dev_id = args.devs[rank % len(args.devs)]
+    cupy.cuda.runtime.setDevice(dev_id)
+    rmm.reinitialize(
+        pool_allocator=True, devices=dev_id, initial_pool_size=args.rmm_init_pool_size
+    )
+
+    # Make cupy use RMM
+    cupy.cuda.set_allocator(rmm.rmm_cupy_allocator)
+
+    df1 = generate_chunk(rank, args.chunk_size, args.n_chunks, "build", args.frac_match)
+    df2 = generate_chunk(rank, args.chunk_size, args.n_chunks, "other", args.frac_match)
+
+    # Let's warmup and sync before benchmarking
+    await distributed_join(args, rank, eps, df1, df2)
+    await barrier(rank, eps)
+
+    if args.cuda_profile:
+        cupy.cuda.profiler.start()
+
+    if args.profile:
+        pr = cProfile.Profile()
+        pr.enable()
+
+    timings = []
+    t1 = clock()
+    df3 = await distributed_join(args, rank, eps, df1, df2, timings)
+    await barrier(rank, eps)
+    took = clock() - t1
+
+    if args.profile:
+        pr.disable()
+        s = io.StringIO()
+        ps = pstats.Stats(pr, stream=s)
+        ps.dump_stats("%s.%0d" % (args.profile, rank))
+
+    if args.cuda_profile:
+        cupy.cuda.profiler.stop()
+
+    return {
+        "bw": sum(t[1] for t in timings) / sum(t[0] for t in timings),
+        "wallclock": took,
+        "throughput": args.n_chunks
+        * len(df2)
+        * len(df2.columns)
+        * sum([t.itemsize for t in df2.dtypes])
+        / took,
+    }
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(description="")
+    parser.add_argument(
+        "--chunks-per-dev",
+        metavar="N",
+        default=1,
+        type=int,
+        help="Number of chunks per device",
+    )
+    parser.add_argument(
+        "-d",
+        "--devs",
+        metavar="LIST",
+        default="0",
+        type=str,
+        help='GPU devices to use (default "0").',
+    )
+    parser.add_argument(
+        "-s",
+        "--server-address",
+        metavar="ip",
+        default=ucp.get_address(),
+        type=str,
+        help="Server address (default `ucp.get_address()`).",
+    )
+    parser.add_argument("-c", "--chunk-size", type=int, default=4, metavar="N")
+    parser.add_argument(
+        "--frac-match",
+        metavar="FRAC",
+        default=0.3,
+        type=float,
+        help="Fraction of rows that matches (default 0.3)",
+    )
+    parser.add_argument(
+        "--net-devices",
+        metavar="LIST",
+        default=None,
+        type=str,
+        help='List of net devices to use, one for each device or "auto"',
+    )
+    parser.add_argument(
+        "--profile",
+        metavar="FILENAME",
+        default=None,
+        type=str,
+        help="Write profile for each worker to `filename.RANK`",
+    )
+    parser.add_argument(
+        "--cuda-profile",
```

Comment:
will this only output if an exception is raise or is there some other output the user should be expecting ?

Reply:
It will write a performance resume such as:
```
cudf merge benchmark
----------------------------
device(s)      | [0, 1, 2, 3, 4, 5, 6, 7]
chunks-per-dev | 1
chunk-size     | 100.00 MB
frac-match     | 0.3
============================
Wall-clock     | 983.40 ms
Bandwidth      | 3.70 GB/s
Throughput     | 26.07 GB/s
```

Reply:
oh, sorry i thought this controlled another report